core/src/stages/generate_pose.cpp

```diff
@@ -76,15 +107,65 @@ void GeneratePose::compute() {
 		return;
 	}
 
-	InterfaceState state(scene);
-	state.properties().set("target_pose", target_pose);
+	const auto& spawn_target_pose = [&](const geometry_msgs::PoseStamped& target_pose) {
+		InterfaceState state(scene);
+		state.properties().set("target_pose", target_pose);
 
-	SubTrajectory trajectory;
-	trajectory.setCost(0.0);
+		SubTrajectory trajectory;
+		trajectory.setCost(0.0);
 
-	rviz_marker_tools::appendFrame(trajectory.markers(), target_pose, 0.1, "pose frame");
+		rviz_marker_tools::appendFrame(trajectory.markers(), target_pose, 0.1, "pose frame");
 
-	spawn(std::move(state), std::move(trajectory));
+		spawn(std::move(state), std::move(trajectory));
+	};
+
+	spawn_target_pose(target_pose);
+
+	// If no dimension samplers were defined, we're done here
+	if (pose_dimension_samplers_.empty())
+		return;
+
+	// Use target pose as seed pose
+	geometry_msgs::PoseStamped seed_pose = target_pose;
+	tf2::Quaternion seed_q, rand_q;
+	tf2::fromMsg(seed_pose.pose.orientation, seed_q);
+	double elapsed_time = 0.0;
+	const auto start_time = std::chrono::steady_clock::now();
+	size_t spawned_solutions = 0;
+	while (elapsed_time < timeout() && spawned_solutions++ < properties().get<size_t>("max_solutions") - 1) {
```

Comment:
Simplify the logic here:
```suggestion
	while (elapsed_time < timeout() && ++spawned_solutions < properties().get<size_t>("max_solutions")) {
```
Also, you should retrieve the `max_solutions` parameter _once_ and _reuse_ it.